File: AlcoholDrive/src/app/domain/repositories/NotificationRepository/Notification.repository.ts
import { Injectable } from "@angular/core";
import { SendAlcResultModel } from "../../model/SendAlcResult.model";

@Injectable({
    providedIn: 'root'
})
export abstract class NotificationRepository {
    /** SlackAPIを取得する */
    abstract fetchSlackAPI(): Promise<void>;
    /** SlackAPIURLを登録する */
    abstract registrySalckAPI(slackAPI: string): Promise<void>;
    /** Slackで検査結果を送信する */
    abstract sendSlack(sendAlcResultModel: SendAlcResultModel): Promise<void>;
}

File: AlcoholDrive-Client/AlcoholDrive-Client/Repository/NotificationRepository.cs
using AlcoholDrive_Client.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AlcoholDrive_Client.Repository {
    public abstract class NotificationRepository {
        /// <summary>
        /// SlackAPIを取得する
        /// </summary>
        /// <returns></returns>
        public abstract string GetSlackAPI();

        /// <summary>
        /// SlackAPIを登録する
        /// </summary>
        /// <param name="slackAPI"></param>
        public abstract void RegistrySlackAPI(string slackAPI);

        /// <summary>
        /// Slackで通知する
        /// </summary>
        /// <param name="sendAlcResult"></param>
        public abstract void SendSlack(SendAlcResult sendAlcResult);
    }
}
